File: core/js/test/vx/web/htmldoc_test.js
'strict mode'

import vx_web_htmldoc from "../../../src/vx/web/htmldoc.js"
import vx_core from "../../../src/vx/core.js"
import vx_test from "../../../src/vx/test.js"
import vx_event from "../../../src/vx/event.js"
import vx_web_html from "../../../src/vx/web/html.js"
import vx_ui_ui from "../../../src/vx/ui/ui.js"

export default class vx_web_htmldoc_test {


  static test_package(context) {
    const testcaselist = vx_web_htmldoc_test.test_cases(context)
    const output = vx_core.f_new(
      vx_test.t_testpackage,
      ":testpkg", "vx/web/htmldoc",
      ":caselist", testcaselist,
      ":coveragesummary", vx_web_htmldoc_test.test_coveragesummary(),
      ":coveragedetail", vx_web_htmldoc_test.test_coveragedetail()
    );
    return output;
  }

  static test_coveragesummary() {
    return vx_core.f_new(
      vx_test.t_testcoveragesummary,
      "testpkg",   "vx/web/htmldoc", 
      "constnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 100, ":tests", 0, ":total", 0), 
      "docnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 100, ":tests", 15, ":total", 15), 
      "funcnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 0, ":tests", 0, ":total", 15), 
      "bigospacenums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 100, ":tests", 0, ":total", 0), 
      "bigotimenums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 100, ":tests", 0, ":total", 0), 
      "totalnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 0, ":tests", 0, ":total", 15), 
      "typenums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 100, ":tests", 0, ":total", 0)
    )
  }

  static test_coveragedetail() {
    return vx_core.f_new(
      vx_test.t_testcoveragedetail,
      "testpkg", "vx/web/htmldoc",
      "typemap",
        vx_core.f_new(
          vx_core.t_intmap
        ),
      "constmap",
        vx_core.f_new(
          vx_core.t_intmap
        ),
      "funcmap",
        vx_core.f_new(
          vx_core.t_intmap,
          "boolean-print<-id-stylesheettext", 0,
          "boolean-remove<-id", 0,
          "boolean-replace<-id-htmltext", 0,
          "boolean-replace<-id-parent-htmltext", 0,
          "boolean-write-stylesheet<-string", 0,
          "boolean-write<-id-attribute-value", 0,
          "boolean-write<-id-htmltext", 0,
          "boolean-write<-id-visible", 0,
          "boolean-write<-stylesheet", 0,
          "boolean-writeevents<-ui", 0,
          "context-read", 0,
          "context-write", 0,
          "string<-id", 0,
          "string<-stylesheet", 0,
          "ui-readstate<-uid", 0
        )
    )
  }

  static test_cases(context) {
    const output = vx_core.f_new(
      vx_test.t_testcaselist
    )
    return output
  }
}

File: core/cpp/src/test/vx/web/htmldoc_test.cpp
#include "../../../main/vx/web/htmldoc.hpp"
#include "../../../main/vx/core.hpp"
#include "../../../main/vx/test.hpp"
#include "../../../main/vx/event.hpp"
#include "../../../main/vx/web/html.hpp"
#include "../../../main/vx/ui/ui.hpp"
#include "htmldoc_test.hpp"

namespace vx_web_htmldoc_test {


  vx_test::Type_testcaselist test_cases(vx_core::Type_context context) {
    vx_core::vx_Type_listany listtestcase;
    vx_test::Type_testcaselist output = vx_core::vx_any_from_any(
      vx_test::t_testcaselist,
      vx_test::t_testcaselist->vx_new_from_list(listtestcase)
    );
    return output;
  }

  vx_test::Type_testcoveragesummary test_coveragesummary() {
    vx_test::Type_testcoveragesummary output = vx_core::vx_new(vx_test::t_testcoveragesummary, {
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/web/htmldoc"),
      vx_core::vx_new_string(":constnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(100), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(0)
      }),
      vx_core::vx_new_string(":docnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(100), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(13), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(13)
      }),
      vx_core::vx_new_string(":funcnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(13)
      }),
      vx_core::vx_new_string(":bigospacenums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(13)
      }),
      vx_core::vx_new_string(":bigotimenums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(13)
      }),
      vx_core::vx_new_string(":totalnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(13)
      }),
      vx_core::vx_new_string(":typenums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(100), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(0)
      })
    });
    return output;
  }

  vx_test::Type_testcoveragedetail test_coveragedetail() {
    vx_test::Type_testcoveragedetail output = vx_core::vx_new(vx_test::t_testcoveragedetail, {
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/web/htmldoc"),
      vx_core::vx_new_string(":typemap"), vx_core::vx_new(vx_core::t_intmap, {
        
      }),
      vx_core::vx_new_string(":constmap"), vx_core::vx_new(vx_core::t_intmap, {
        
      }),
      vx_core::vx_new_string(":funcmap"), vx_core::vx_new(vx_core::t_intmap, {
        vx_core::vx_new_string(":boolean-print<-id-stylesheettext"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":boolean-replace<-id-htmltext"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":boolean-replace<-ui-htmltext"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":boolean-write-stylesheet<-string"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":boolean-write<-id-htmltext"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":boolean-write<-id-visible"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":boolean-write<-stylesheet"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":boolean-writeevents<-ui"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":context-read"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":context-write"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":string<-id"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":string<-stylesheet"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":ui-readstate<-uid"), vx_core::vx_new_int(0)
      })
    });
    return output;
  }

  vx_test::Type_testpackage test_package(vx_core::Type_context context) {
    vx_test::Type_testcaselist testcaselist = vx_web_htmldoc_test::test_cases(context);
    vx_test::Type_testcoveragesummary testcoveragesummary = vx_web_htmldoc_test::test_coveragesummary();
    vx_test::Type_testcoveragedetail testcoveragedetail = vx_web_htmldoc_test::test_coveragedetail();
    vx_test::Type_testpackage output = vx_core::vx_new(vx_test::t_testpackage, {
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/web/htmldoc"),
      vx_core::vx_new_string(":caselist"), testcaselist,
      vx_core::vx_new_string(":coveragesummary"), testcoveragesummary,
      vx_core::vx_new_string(":coveragedetail"), testcoveragedetail
    });
    return output;
  }

}
